src/plugins/discover/public/application/main/components/field_stats_table/types.ts

```diff
@@ -164,4 +164,9 @@ export interface FieldStatisticsTableProps {
    * Search session id to save to or restore from
    */
   searchSessionId?: string;
+  /**
+   * If table should should accept text based query
+   * isPlainRecord is true when in ES|QL mode
+   */
+  isPlainRecord?: boolean;
```

Comment:
@davismcphee here https://github.com/elastic/kibana/pull/183108 is changing the `isPlainRecord` to `isEsqlMode `. Let's be proactive and use `isEsqlMode` here instead